deps/rabbit/src/rabbit_ff_controller.erl

```diff
@@ -1178,9 +1178,32 @@ this_node_first(Nodes) ->
       Ret :: term() | {error, term()}.
 
 rpc_call(Node, Module, Function, Args, Timeout) ->
+    SleepBetweenRetries = 5000,
+    T0 = erlang:monotonic_time(),
     try
         erpc:call(Node, Module, Function, Args, Timeout)
     catch
+        %% In case of `noconnection' with `Timeout'=infinity, we don't retry
+        %% at all. This is because the infinity "timeout" is used to run
+        %% callbacks on remote node and they can last an indefinite amount of
+        %% time, for instance, if there is a lot of data to migrate.
+        error:{erpc, noconnection} = Reason
+          when is_integer(Timeout) andalso Timeout > SleepBetweenRetries ->
+            ?LOG_ERROR(
+               "Feature flags: no connection to node `~ts`; "
+               "retrying in ~b milliseconds",
+               [Node, SleepBetweenRetries],
+               #{domain => ?RMQLOG_DOMAIN_FEAT_FLAGS}),
+            timer:sleep(SleepBetweenRetries),
+            T1 = erlang:monotonic_time(),
+            TDiff = erlang:convert_time_unit(T1 - T0, native, millisecond),
+            Remaining = Timeout - TDiff,
+            Timeout1 = erlang:max(Remaining, 0),
+            case Timeout1 of
+                0 -> {error, Reason};
```

Comment:
Should this be:

```
erlang:raise(error, {erpc, noconnection}, Stack);
```

(after capturing `Stack` in the `catch`)

...instead, so that we don't change what happens when `error:{erpc, noconnection}` is raised?

Reply:
The function always caught exceptions from `erpc:call/5`. Before this patch, it would log the error and return `{error, {erpc, noconnection}}`.

The callers don't catch anything and expect an error return value.

Reply:
Look at that, I just had to expand the diff a bit to see what used to happen. Carry on!